packages/story-editor/src/app/rightClickMenu/useLayerSelect.js

```diff
@@ -70,21 +72,55 @@ function useLayerSelect({ menuItemProps, menuPosition, isMenuOpen }) {
       return [];
     }
     const clickedPoint = new SAT.Vector(x, y);
-    const intersectingElements = currentPage.elements
-      .map((element) => {
-        const { id, rotationAngle } = element;
-        const node = nodesById[id];
-        if (!node) {
-          return null;
-        }
-        const { x: elX, y: elY, width, height } = node.getBoundingClientRect();
-        const elementP = createPolygon(rotationAngle, elX, elY, width, height);
-        return SAT.pointInPolygon(clickedPoint, elementP) ? element : null;
-      })
-      .filter((el) => el);
+    const bgElement = currentPage.elements.find(
+      ({ isBackground }) => isBackground
+    );
+    const bgNode = nodesById[bgElement.id];
+    const {
+      x: bgX,
+      y: bgY,
+      width: bgWidth,
+      height: bgHeight,
+    } = bgNode.getBoundingClientRect();
+    const bgPolygon = new SAT.Box(
+      new SAT.Vector(bgX, bgY),
+      bgWidth,
+      bgHeight
+    ).toPolygon();
+    // The background element is offset by a certain amount, so add that back in for
+    // other elements when resolving their position relative to it.
+    const elementYOffset = -getBox(bgElement).y;
+    const elementsWithPolygons = currentPage.elements.map((element) => {
+      if (element.isBackground) {
+        return { element, polygon: bgPolygon };
+      }
+      const { id, rotationAngle } = element;
+      const node = nodesById[id];
+      if (!node) {
+        return null;
+      }
+      const box = getBox(element);
+      // Note that we place the box at the center coordinate. We do this for the angle
+      // to apply correctly. We correct for this offset with `setOffset`` later.
+      const center = new SAT.Vector(
+        bgPolygon.pos.x + box.x + box.width / 2,
+        bgPolygon.pos.y + elementYOffset + box.y + box.height / 2
+      );
+      const elementBox = new SAT.Box(center, box.width, box.height);
+      const polygon = elementBox.toPolygon();
+      const offset = new SAT.Vector(-box.width / 2, -box.height / 2);
+      polygon.setOffset(offset);
+      polygon.setAngle((rotationAngle * Math.PI) / 180);
+      return { element, polygon };
+    });
```

Comment:
Does this require more comments? I feel it's pretty straight-forward except for the center/offset part. The [SAT documentation is decent](https://github.com/jriecken/sat-js#satvector-aliased-as-satv), so could link to that? It would be a fragile link though.